src/datalink/mod.rs

```diff
@@ -76,7 +76,7 @@ pub fn datalink_channel(network_interface: &NetworkInterface,
 /// Structure for sending packets at the data link layer. Should be constructed using
 /// datalink_channel().
 pub struct DataLinkSender {
-    dlsi: backend::DataLinkSenderImpl
+    sender: backend::DataLinkSenderImpl
```

Comment:
I think this would be better named `backend`, and the same with the `DataLinkReceiver` struct.